### ISSUE TITLE
Order of ORDERED enums not considered in attribute forms

### ISSUE THREAD

**Issue by webrian** [opengisch/QgisModelBaker#940]
The order of an ORDERED enumeration is not considered by Model Baker.

A simple INTERLIS model with a ordered enumeration:

```
INTERLIS 2.3;

MODEL SimpleTable (de)
AT "mailto:agi@bd.so.ch"
VERSION "2024-05-02"  =

  DOMAIN

    LKoord = COORD 2460000.000 .. 2870000.000 [INTERLIS.M], 1045000.000 .. 1310000.000 [INTERLIS.M] ,ROTATION 2 -> 1;

    SimpleArt = (
      ErsteArt,
      ZweiteArt,
      DritteArt,
      VierteArt,
      FuenfteArt,
      SechsteArt
    ) ORDERED;

  TOPIC SimpleTopic =

    CLASS Table =
      Anzahl : 1 .. 1000;
      Bezeichnung : MTEXT*64;
      Geometrie : MANDATORY SimpleTable.LKoord;
      Art : SimpleTable.SimpleArt;
    END Table;

  END SimpleTopic;

END SimpleTable.
```

This model is implemented in the (PG) database as follows:

![image](https://github.com/opengisch/QgisModelBaker/assets/980073/8fd36cf9-3dd0-4505-8e40-da7c59521e4c)

The column `itfcode` represents the order as defined in the data model.

Creating the QGIS project with Model Baker the enum values are ordered in an alphabetical way:

![Bildschirmfoto vom 2024-06-25 15-23-30_dialog](https://github.com/opengisch/QgisModelBaker/assets/980073/25458b4b-fccc-4b2c-9b2c-36ec4a60d692)

I think from a user's point of view it would be preferable if the order of the values corresponded to the data model.

See also the discussion in the INTERLIS forum: https://interlis.discourse.group/t/ordered-enumerations/175

**Comment by webrian** [comment]
Currently the `seq` column does not correspond to the `itf_code`. Not sure if this is on purpose or an issue with ili2db?

**Comment by signedav** [comment]
Yes you are right. I have to clarify that.

**Comment by gacarrillor** [comment]
> Shouldn't the `seq` giving the order?
> 
> Anyway, I think this would require a QGIS change. Maybe this is something to consider [qgis/QGIS#48191](https://github.com/qgis/QGIS/issues/48191)
> 
> Still Model Baker should then set the sorting to the `seq` (or `itf_code`)

See https://github.com/claeis/ili2db/issues/526#event-11155599672

**Comment by gacarrillor** [comment]
From the [ili2db docs](https://github.com/claeis/ili2db/blob/master/docs/ili2db.rst):

`seq Definiert die Reihenfolge der Aufzählelemente.`

**Comment by signedav** [comment]
Since 3.42 QGIS supports ordering on Value Relation widgets https://qgis.org/project/visual-changelogs/visualchangelog342/#feature-additional-sorting-options-on-value-relation-widget 

This would be an alternative approach to use the Value Relation instead of the Relation Reference widgets for enumerations. For Catalogues I would still prefer to have the Relation Reference to be able to open the form for more information. And for all other lookups of course, since we might need to add new features or select on map etc.

But for Enumerations the Value Relation would be sufficient. What do you think  @gacarrillor ? And @webrian you still have an opinion on that?

In the end, I would although prefer the QGIS implementation, since we are able then to sort all the other lookups as well. But better Value Relations than nothing.

**Comment by gacarrillor** [comment]
> Since 3.42 QGIS supports ordering on Value Relation widgets https://qgis.org/project/visual-changelogs/visualchangelog342/#feature-additional-sorting-options-on-value-relation-widget
> 
> This would be an alternative approach to use the Value Relation instead of the Relation Reference widgets for enumerations. For Catalogues I would still prefer to have the Relation Reference to be able to open the form for more information. And for all other lookups of course, since we might need to add new features or select on map etc.
> 
> But for Enumerations the Value Relation would be sufficient. What do you think [@gacarrillor](https://github.com/gacarrillor) ? And [@webrian](https://github.com/webrian) you still have an opinion on that?
> 
> In the end, I would although prefer the QGIS implementation, since we are able then to sort all the other lookups as well. But better Value Relations than nothing.

No strong opinion, but looks good to me.

**Comment by webrian** [comment]
> Since 3.42 QGIS supports ordering on Value Relation widgets https://qgis.org/project/visual-changelogs/visualchangelog342/#feature-additional-sorting-options-on-value-relation-widget
> 
> This would be an alternative approach to use the Value Relation instead of the Relation Reference widgets for enumerations. For Catalogues I would still prefer to have the Relation Reference to be able to open the form for more information. And for all other lookups of course, since we might need to add new features or select on map etc.
> 

Yes, I still do have an opinion on that .. ;) If I understand you correctly, you would implement all enumerations with a value relation widget instead of a relation widget? That would be quite a change, wouldn't it?
As you maybe remember I've made some bad experiences with GeoPackage and (too) many relations. There my workaround were value relation widgets as well. From this point I agree with a change to value relation widgets for enums.
